engine/src/main/java/com/datatorrent/stram/engine/Node.java

```diff
@@ -317,9 +317,17 @@ public void removeSinks(Map<String, Sink<Object>> sinks)
   protected ProcessingMode PROCESSING_MODE;
   protected volatile boolean shutdown;
 
-  public void shutdown()
+  /**
+   * If shutdown the node by setting alive to false. If processEndWindow is true, then after shutting
+   * down endWindow is called and END_STREAM tuple is sent to downstream operators.
+   *
+   * @param processEndWindow
+   */
+  public void shutdown(boolean processEndWindow)
```

Comment:
You will have to make changes in StreamingAppMasterService, *forcedShutdown* variable will mark the application as *failed* instead of *Finished*